source/codegen/metadata/nirfmxspecan_restricted/functions.py

```diff
@@ -0,0 +1,98 @@
+functions = {
+    'CacheResult': {
+        'parameters': [
+            {
+                'direction': 'in',
+                'grpc_name': 'instrument',
+                'name': 'instrumentHandle',
+                'type': 'niRFmxInstrHandle',
+            },
+            {
+                'direction': 'in',
+                'name': 'selectorString',
+                'type': 'char[]'
+            },
+            {
+                'direction': 'in',
+                'name': 'selectorStringOutSize',
+                'type': 'int32'
+            },
+            {
+                'direction': 'out',
+                'name': 'selectorStringOut',
+                'size': {
+                    'mechanism': 'passed-in',
```

Comment:
How do we know when to use 'passed-in' vs. 'ivi-dance' for the mechanism?

Reply:
I found it the hard way when my test failed. I guess we just have to know the API - did not help that this one is private and not documented. Having said that, most APIs support ivi-dance and we should first use that and see if it works.